api/mongowriter.go

```diff
@@ -49,3 +95,35 @@ func (mw *MongoWriter) WriteHeader(statusCode int) {
 	}
 	mw.w.WriteHeader(statusCode)
 }
+
+// ErrorBuffer returns the data stored in the error buffer.
+func (mw *MongoWriter) ErrorBuffer() []byte {
+	return mw.ew.Buffer.Bytes()
+}
+
+// ErrorStatus returns the status code with which the last call errored out,
+// if any.
+func (mw *MongoWriter) ErrorStatus() int {
+	return mw.ew.Status
+}
+
+// FailedWithWriteConflict informs us whether the MongoWriter received a MongoDB
+// WriteConflict error.
+func (mw *MongoWriter) FailedWithWriteConflict() bool {
+	return mw.ew.Status != 0 && strings.Contains(mw.ew.Buffer.String(), writeConflictErrMsg)
+}
+
+// Header implementation.
+func (w *bufferResponseWriter) Header() http.Header {
+	return http.Header{}
+}
+
+// Write implementation.
+func (w *bufferResponseWriter) Write(b []byte) (int, error) {
+	return w.Buffer.Write(b)
```

Comment:
This is something we definitely should keep in mind. We were using a stdlib TimeoutHandler in skyd once which also buffered the whole response without us knowing and as you can imagine downloading a 10GiB would cause 10GiB of RAM usage which wasn't ideal.
So this is something that only works as long as responses are reasonably small.

Reply:
Yeah me neither. Just something to remember in case we ever run into it. No immediate action item here.